module/VuFind/src/VuFind/View/Helper/Root/SearchMemory.php

```diff
@@ -89,6 +91,15 @@ public function getLastSearchLink($link, $prefix = '', $suffix = '')
             $url = $urlHelper($lastSearch->getOptions()->getSearchAction())
                 . $lastSearch->getUrlQuery()->getParams(false);
 
+            // Try to append page number and page size from optional params saved in results object
```

Comment:
If you take my proxy suggestion above, this code might not be necessary. If this code IS necessary, it might be better to leverage the URL Query Helper to add page/limit, like:

```
$url = $urlHelper($lastSearch->getOptions()->getSearchAction());
$queryHelper = $lastSearch->getUrlQuery();
// Try to append page number and page size from optional params saved in results object
$optionalParameters = $lastSearch->getOptionalParameters();
if (!empty($optionalParameters['page'])) {
    $queryHelper->setPage($optionalParameters['page']);
}
if (!empty($optionalParameters['limit'])) {
    $queryHelper->setLimit($optionalParameters['limit']);
}
$url .= $queryHelper->getParams(false);
```